src/cs/Ssh/TaskChain.cs

```diff
@@ -49,6 +49,16 @@ public async Task RunInSequence(Func<Task> task, Action<Exception> onError, Func
 		try
 		{
 			await semaphore.WaitAsync(cancellation).ConfigureAwait(false);
+			
+			if (runInSequenceTask != null && 
+				(runInSequenceTask.IsCanceled || runInSequenceTask.Exception != null))
+			{
+				// If one task in the sequence is cancelled we have to reset runInSequenceTask 
+				// so that all subeequent queueing will succeed.
+				runInSequenceTask = null;
+			}
+
+			cancellation.ThrowIfCancellationRequested();
```

Comment:
nit: no need because `semaphore.WaitAsync` should have thrown already if `cancellation` was cancelled.